mindsdb/integrations/handlers/slack_handler/slack_handler.py

```diff
@@ -394,14 +399,21 @@ def _process_websocket_message(client: SocketModeClient, request: SocketModeRequ
                 return
 
             payload_event = request.payload['event']
-            if payload_event['type'] not in ('message', 'app_mention'):
+
+            if payload_event['type'] == 'message' and payload_event['channel_type'] != 'im':
+                # Avoid responding to messages in channels
+                return
+
+            if payload_event['type'] == 'app_mention' and self._get_my_user_id() not in payload_event['text']:
```

Comment:
@ea-rus Yes, I think you are right. Let me update the logic to store this in handler storage and reduce the number of API calls.

This condition is mainly to handle situations where multiple apps have been added to the same channel. Without this, if a message mentions a different app (and not the one connected to the bot), it will still respond.

Reply:
@ea-rus I have implemented the caching logic via handler storage to reduce API calls being made with every request. I have also stored the channels data in the cache, because otherwise, for large work-spaces, we could run into API rate limits.